trinity/rpc/format.py

```diff
@@ -96,7 +95,11 @@ def to_receipt_response(receipt: ReceiptAPI,
         ],
         "logsBloom": format_bloom(receipt.bloom),
         "root": encode_hex(receipt.state_root),
-        "to": encode_hex(transaction.to),
+        "to": apply_formatter_if(
+            is_address,
+            to_checksum_address,
+            encode_hex(transaction.to)
+        ),
```

Comment:
Having the encode in the argument tripped me for a second. I took a stab at an alternative.

```suggestion
        "to": compose(
            apply_formatter_if(
                is_address,
                to_checksum_address
            ),
            encode_hex,
        )(transaction.to),
```

Not really sure it's any better.

Reply:
It was already used that way in `transaction_to_dict` and I personally find it slightly more readable but I'd be fine either way.